ci/scripts/cpp_build.sh

```diff
@@ -283,9 +283,10 @@ else
 fi
 
 if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
-  time meson install
+  time meson compile -j ${ARROW_BUILD_PARALLEL:-$[${n_jobs} + 1]}
+  meson install
 else
-  export CMAKE_BUILD_PARALLEL_LEVEL=${CMAKE_BUILD_PARALLEL_LEVEL:-$[${n_jobs} + 1]}
+  export ARROW_BUILD_PARALLEL=${ARROW_BUILD_PARALLEL:-$[${n_jobs} + 1]}
```

Comment:
CMake doesn't know what `ARROW_BUILD_PARALLEL` is, we should still use `CMAKE_BUILD_PARALLEL_LEVEL` which we can define outside of the if `: ${ARROW_BUILD_PARALLEL:=$[${n_jobs} + 1]}`
```suggestion
  export CMAKE_BUILD_PARALLEL_LEVEL=${ARROW_BUILD_PARALLEL}
```

Reply:
Nice catch - done!